neo/SmartContract/tests/test_smart_contract.py

```diff
@@ -45,6 +45,8 @@ def test_a_run_sc(self):
 
         block = Helper.AsSerializableWithType(hexdata, 'neo.Core.Block.Block')
 
-        result = self._blockchain.Persist(block)
+        result = False
+        with BlockchainFixtureTestCase.MPPersist():
```

Comment:
can you explain the need for this monkey patched persist vs just persisting directly?
if it is about capturing the execution success status we could mock `ExecutionCompleted()`
https://github.com/CityOfZion/neo-python/blob/f08dd6b90a9acc7f2f4b73f2977a316c8899e7af/neo/SmartContract/ApplicationEngine.py#L317 and read the `success` argument passed to the function. Or if the SC emits any events subscribe to the event handler and assert that we get the expected output emitted here
https://github.com/CityOfZion/neo-python/blob/f08dd6b90a9acc7f2f4b73f2977a316c8899e7af/neo/SmartContract/ApplicationEngine.py#L322-L323

My main concern is the code duplication by the `MPersist()` knowing that these are critical areas (and subject to optimisation).

Reply:
This is nothing I invented, I just moved it to `BlockchainFixtureTestCase`. In the old code it was located within the `TestLevelDBBlockchain`. I just made it work with the current implementation. I can still change it if you really want to get rid of it.